store/postgres/src/entities.rs

```diff
@@ -1445,16 +1445,32 @@ impl Table {
         index: &AttributeIndexDefinition,
     ) -> Result<usize, StoreError> {
         let (index_type, index_operator, jsonb_operator) = match index.field_value_type {
-            ValueType::Boolean
-            | ValueType::BigInt
+            ValueType::Boolean => {
+                // There's no point in having an index on a boolean attribute;
+                // since there are only two possible values, the index will
+                // not be very selective, and Postgres will just scan the
+                // table anyway
+                return Ok(1);
+            }
```

Comment:
If we adopt a query costing model that boils down to `constant + size(result)`, then a good property to have is that cheap queries have small result sizes and expensive queries have large result sizes. Having indexes on booleans helps maintain that property.

If an entity has evenly distributed `false` and `true`, then a full scan will be done but will generate a proportionally large result set, so we're good. However if the table for example has only a few `true` values, then the index is important. Without the index, selecting the `true` rows will cause a full scan but return a small result set, breaking the property. With an index, we'd use it and therefore have a cheap query for a small result set, maintaining the property.

Reply:
You are right that this is not the right way to address our overindexing, and we should look more to see what indexes are/are not used before removing some. I took that part out of the PR.